components/LocalePicker.php

```diff
@@ -142,7 +150,29 @@ protected function makeLocaleUrlFromPage($locale)
             $router = new RainRouter;
 
             $params = $this->getRouter()->getParameters();
-
+                        
+            /**
+             * @event translate.params.translateParams
+             * Enables manipulating the URL parameters
+             *
+             * You will have access to the page object, the old and new locale and the URL parameters.
+             *
+             * Example usage:
+             *
+             *     Event::listen('translate.params.translateParams', function($page, $params, $oldLocale, $newLocale) {
+             *        if ($page->baseFileName == 'your-page-filename') {
+             *             return YourModel::translateParams($params, $oldLocale, $newLocale);
+             *         }
+             *     });
+             *
+             */            
+            $translatedParams = Event::fire('translate.params.translateParams', 
```

Comment:
One final change, please change the event name to `translate.localePicker.translateParams` everywhere. Thanks!